source/material_model/melt_boukare.cc

```diff
@@ -0,0 +1,1252 @@
+    {
+      const double thermal_energy = endmember_thermal_energy(temperature, i);
+      const double heat_capacity = endmember_molar_heat_capacity(reference_temperature, i);
+      const double thermal_pressure = reference_thermal_expansivities[i] * reference_bulk_moduli[i] / heat_capacity * thermal_energy;
+
+      return thermal_pressure;
+    }
+
+
+    template <int dim>
+    double
+    MeltBoukare<dim>::
+    endmember_enthalpy_thermal_addition (const double temperature,
+                                         const unsigned int i) const
+    {
+      const double addition = reference_specific_heats[i] * temperature
+                              + 0.5 * specific_heat_linear_coefficients[i] * std::pow(temperature, 2.)
+                              - specific_heat_second_coefficients[i] / temperature
+                              + 2. * specific_heat_third_coefficients[i] * sqrt(temperature)
+                              - (reference_specific_heats[i] * reference_temperature
+                                 + 0.5 * specific_heat_linear_coefficients[i] * std::pow(reference_temperature, 2.)
+                                 - specific_heat_second_coefficients[i] / reference_temperature
+                                 + 2.0 * specific_heat_third_coefficients[i] * sqrt(reference_temperature));
+
+      return addition;
+    }
+
+
+    template <int dim>
+    double
+    MeltBoukare<dim>::
+    endmember_entropy_thermal_addition (const double temperature,
+                                        const unsigned int i) const
+    {
+      const double addition = reference_specific_heats[i] * std::log(temperature)
+                              + specific_heat_linear_coefficients[i] * temperature
+                              - 0.5 * specific_heat_second_coefficients[i] / std::pow(temperature, 2.)
+                              - 2.0 * specific_heat_third_coefficients[i] / sqrt(temperature)
+                              - (reference_specific_heats[i] * std::log(reference_temperature)
+                                 + specific_heat_linear_coefficients[i] * reference_temperature
+                                 - 0.5 * specific_heat_second_coefficients[i] / std::pow(reference_temperature, 2.)
+                                 - 2.0 * specific_heat_third_coefficients[i] / sqrt(reference_temperature));
+
+      return addition;
+    }
+
+
+    template <int dim>
+    void
+    MeltBoukare<dim>::
+    convert_composition_to_fraction_of_endmembers (const double temperature,
+                                                   const double molar_Fe_in_solid,
+                                                   const double molar_Fe_in_melt,
+                                                   const std::vector<double> &endmember_gibbs_energies,
+                                                   std::vector<double> &endmember_mole_fractions_per_phase,
+                                                   double &f_pv) const
```

Comment:
In the header the last parameter is called `molar_bridgmanite_in_solid`, which is easier to understand. Please unify.